src/contracts/CrossChainForwarder.sol

```diff
@@ -261,6 +279,44 @@ contract CrossChainForwarder is OwnableWithGuardian, ICrossChainForwarder {
     _disableBridgeAdapters(bridgeAdapters);
   }
 
+  /// @inheritdoc ICrossChainForwarder
+  function updateOptimalBandwidthByChain(
+    OptimalBandwidthByChain[] memory optimalBandwidthByChain
+  ) external onlyOwner {
+    _updateOptimalBandwidthByChain(optimalBandwidthByChain);
+  }
+
+  /**
+   * @notice method to get a shuffled array of forwarder bridge adapters configurations
+   * @param destinationChainId id of the destination chain to get the adapters that communicate to it
+   * @return a shuffled array of the forwarder configurations for a destination chain
+   */
+  function _getShuffledBridgeAdaptersByChain(
+    uint256 destinationChainId
+  ) internal view returns (ChainIdBridgeConfig[] memory) {
+    uint256 optimalBandwidth = _optimalBandwidthByChain[destinationChainId];
+    ChainIdBridgeConfig[] storage forwarderAdapters = _bridgeAdaptersByChain[destinationChainId];
+
+    // If configured optimal bandwidth for a destination network are set to 0 or are bigger than current adapters,
+    // it will use all the adapters available. This way there would be no way of breaking forwarding communication
+    // by setting wrong configuration.
+    if (optimalBandwidth == 0 || optimalBandwidth >= forwarderAdapters.length) {
+      return forwarderAdapters;
+    }
+
+    uint256[] memory shuffledIndexes = Math.shuffleArray(Math.generateIndexArray(optimalBandwidth));
```

Comment:
Why this (and all follow-ups) is not a fixed-size array? We know the size with `optimalBandwidth`

Reply:
Wouldn't it be a problem on the return then?

Reply:
Mmh, I think not, but could be. Better to check